core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1387,6 +1387,27 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   }
 
   /**
+    * Return a list of file paths that are added to resources.
+    * If file paths are provided, return the ones that are added to resources.
+    */
+  def listFiles(files: Seq[String] = Seq.empty[String]): Seq[String] = {
```

Comment:
@rxin Just one concern about thiss one. It is possible that users just invoked listFiles or listJars directly with sparkContext. Do we want to provide filtering for this case? Right now, I have a [test case](https://github.com/xwu0226/spark/blob/21b092ab84b22abec93fde1fc1ca177db68d9f0d/core/src/test/scala/org/apache/spark/SparkContextSuite.scala#L159-L176) that covers this case.